provider/pkg/schema/convert.go

```diff
@@ -58,6 +58,18 @@ func (c *sdkToCfnConverter) sdkTypedValueToCfn(spec *pschema.TypeSpec, v interfa
 		return c.sdkObjectValueToCfn(typName, v)
 	}
 
+	if spec.OneOf != nil {
+		for _, item := range spec.OneOf {
+			converted := c.sdkTypedValueToCfn(&item, v)
+			// If the conversion resulted in a different value, return it.
+			// We should probably fail if we can't convert rather than considering change a success.
+			// This is likely to happen if one of the types is any.
+			if !reflect.DeepEqual(converted, v) {
+				return converted
```

Comment:
I find this code a bit surprising. Why return the first successful `OneOf`, what if there are more? And when the conversion results in an identical object, is that not still a valid CFN object?

Reply:
Yeah, the whole way this conversion works is questionable. Essentially a failure to convert is just returning the input back as the result.

This was the smallest possible change I could find without reworking all the functions. We would need some way to check if the input is compatible with the target in a "tryConvert" method then return the first successful result. Then again, technically we're using `oneOf` here, not `anyOf` so we should be failing if more than one matches ... but the second case in this `oneOf` is any - so that would always fail.